tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/views/source_files/source_files_container_test.ts

```diff
@@ -0,0 +1,237 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+/**
+ * Unit tests for the the Source Files component and container.
+ */
+import {CommonModule} from '@angular/common';
+import {TestBed} from '@angular/core/testing';
+import {By} from '@angular/platform-browser';
+
+import {Store} from '@ngrx/store';
+import {provideMockStore, MockStore} from '@ngrx/store/testing';
+
+import {DebuggerComponent} from '../../debugger_component';
+import {DebuggerContainer} from '../../debugger_container';
+import {DataLoadState, State} from '../../store/debugger_types';
+import {createDebuggerState, createState} from '../../testing';
+import {AlertsModule} from '../alerts/alerts_module';
+import {ExecutionDataModule} from '../execution_data/execution_data_module';
+import {InactiveModule} from '../inactive/inactive_module';
+import * as loadMonacoShim from '../source_code/load_monaco_shim';
+import {StackTraceModule} from '../stack_trace/stack_trace_module';
+import {
+  getFocusedSourceFileContent,
+  getFocusedSourceLineSpec,
+} from '../../store';
+import {TimelineModule} from '../timeline/timeline_module';
+import {SourceFilesContainer} from './source_files_container';
+import {SourceFilesModule} from './source_files_module';
+
+/** @typehack */ import * as _typeHackStore from '@ngrx/store';
+
+describe('Source Files Container', () => {
+  let store: MockStore<State>;
+  let dispatchSpy: jasmine.Spy;
+
+  class FakeRange {
+    constructor(
+      readonly startLineNumber: number,
+      readonly startColumn: number,
+      readonly endLineNumber: number,
+      readonly endColumn: number
+    ) {}
+  }
+
+  let loadMonacoSpy: jasmine.Spy;
+  // TODO(cais): Explore better typing by depending on 3rd-party libraries.
+  let editorSpy: jasmine.SpyObj<any>;
+  let monaco: any;
+  function setUpMonacoFakes() {
+    async function fakeLoadMonaco() {
```

Comment:
Hmm, another way to have sane test and what not would be to inject the monaco editor and provide a shim implementation.